testing/trino-testing/src/main/java/io/trino/testing/AbstractTestEngineOnlyQueries.java

```diff
@@ -6694,6 +6694,47 @@ SELECT a(10)
                 .hasMessage("line 3:8: Recursive language functions are not supported: a(integer):integer");
     }
 
+    // regression test for https://github.com/trinodb/trino/issues/19997
+    @Test
+    public void testIssue19997()
+    {
+        assertThat(query("""
+                WITH FUNCTION format_topn(input map<varchar, bigint>)
+                    RETURNS VARCHAR
+                    NOT DETERMINISTIC
+                    RETURNS NULL ON NULL INPUT
+                    BEGIN
+                    DECLARE freq_separator VARCHAR DEFAULT '=';
+                    DECLARE entry_separator VARCHAR DEFAULT ', ';
+                    RETURN array_join(transform(
+                            reverse(array_sort(transform(
+                                transform(
+                                    map_entries(input),
+                                    r -> cast(r AS row(key varchar, value bigint))
+                                ),
+                                r -> cast(row(r.value, r.key) AS row(value bigint, key varchar)))
+                            )),
+                            r -> r.key || freq_separator || cast(r.value as varchar)),
+                        entry_separator);
+                    END
+                WITH data AS (
+                    SELECT lpad('', 3, chr(65+(s.num / 3))) AS value FROM table(sequence(start=>1, stop=>10)) AS s(num)
```

Comment:
Is there a simpler repro? This query is doing a lot of things that seem unrelated to the bug

Reply:
Here's a simpler repro:

```sql
WITH FUNCTION get(input row(varchar))
	RETURNS varchar
	RETURN input[1]
SELECT get(row('abc'))
```

Reply:
i wanted to have a regression test for the issue. 

yes, i can further try to find a smaller repo

Reply:
> Here's a simpler repro:

thanks added

i can delete the more verbose test case
but note that it potentially exposes some other problem as well https://github.com/trinodb/trino/pull/20181#discussion_r1433188325

Reply:
Yes, please remove it. It's not needed for this issue. If necessary, it (or a simpler variant) can be added when the other issue is addressed.